pkg/netstore/netstore.go

```diff
@@ -57,31 +70,70 @@ func (s *store) Get(ctx context.Context, mode storage.ModeGet, addr swarm.Addres
 				go s.recoveryCallback(addr, targets)
 				return nil, ErrRecoveryAttempt
 			}
-			stamp, err := ch.Stamp().MarshalBinary()
-			if err != nil {
-				return nil, err
-			}
-
-			putMode := storage.ModePutRequest
-			if mode == storage.ModeGetRequestPin {
-				putMode = storage.ModePutRequestPin
-			}
-
-			cch, err := s.validStamp(ch, stamp)
-			if err != nil {
-				// if a chunk with an invalid postage stamp was received
-				// we force it into the cache.
-				putMode = storage.ModePutRequestCache
-				cch = ch
-			}
-
-			_, err = s.Storer.Put(ctx, putMode, cch)
-			if err != nil {
-				return nil, fmt.Errorf("netstore retrieve put: %w", err)
-			}
+			s.put(ch, mode)
 			return ch, nil
 		}
 		return nil, fmt.Errorf("netstore get: %w", err)
 	}
 	return ch, nil
 }
+
+// put will store the chunk into storage asynchronously
+func (s *store) put(ch swarm.Chunk, mode storage.ModeGet) {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+
+		select {
+		case <-s.sCtx.Done():
```

Comment:
you can create a new context in this function and when the quit chan is closed, the context can be canceled.